app/src/lib/git/diff.ts

```diff
@@ -38,6 +38,14 @@ const MaxDiffBufferSize = 268435441
  */
 const MaxReasonableDiffSize = 3000000
 
+/**
+ * Where `MaxReasonableDiffSize` is a soft limit, and `MaxDiffBufferSize`
+ * is an absolute limit, this is the MAX number of bytes to read from the
+ * buffer before _assuming_ the current buffer being read is `MaxDiffBufferSize`.
+ * This is done so that the UI isn't waiting for the entire buffer to be read.
+ */
+const MaxBytesToRead = MaxDiffBufferSize / 8 //~32MB
```

Comment:
Couldn't we just use MaxReasonableDiffSize instead? Do we care about that small zone in between `MaxBytesToRead` and `MaxReasonableDiffSize`?

Reply:
If we used `MaxReasonableDiffSize` instead, that would lead to very relatively small diffs being classified as too large (~3MB).  Having a separate value allows us to classify the diff as potentially too large with some wiggle room (up to 32MB) to display it.

Reply:
I also want to point out that this is being used over in #4050 and I'm using each of these values to determine when to show a diff and when to mark it as not renderable.

Reply:
> If we used MaxReasonableDiffSize instead, that would lead to very relatively small diffs being classified as too large (~3MB). Having a separate value allows us to classify the diff as potentially too large with some wiggle room (up to 32MB) to display it.

What's that middle ground state giving us? The way I look at it there's three states of a diff from the perspective of the app. There's a diff that we'll always show (<3mb), there's diffs that we won't show by default but the user can explicitly tell us to load (>=3mb) and then there's diffs which we will never be able to represent due to limitations in node (MaxDiffBufferSize).

So there's a soft limit and a hard limit. When we go beyond the soft limit we should exit quickly in this method and show the user that we believe the diff to be unreasonably large. If they choose to try loading it anyway we should call this method again and tell it not to care about the soft limit. What am I missing?